src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecurityProtocolFactory.cs

```diff
@@ -686,6 +693,31 @@ public virtual void OnClose(TimeSpan timeout)
             }
         }
 
+        public virtual Task OnCloseAsync(TimeSpan timeout)
+        {
+            return OnCloseAsyncInternal(timeout);
+        }
+
+        private async Task OnCloseAsyncInternal(TimeSpan timeout)
+        {
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            if (!_actAsInitiator)
+            {
+                foreach (SupportingTokenAuthenticatorSpecification spec in _channelSupportingTokenAuthenticatorSpecification)
+                {
+                    await SecurityUtils.CloseTokenAuthenticatorIfRequiredAsync(spec.TokenAuthenticator, timeoutHelper.RemainingTime());
```

Comment:
>timeoutHelper.RemainingTime() [](start = 104, length = 29)

Is it worth converting to CancellationToken's? You can instantiate a single one at the beginning and reuse.